src/utils/cargo_metadata.rs

```diff
@@ -79,11 +79,15 @@ pub(crate) struct Package {
 }
 
 impl Package {
-    pub(crate) fn is_library(&self) -> bool {
-        match self.targets[0].kind.as_slice() {
+    pub(crate) fn library_target(&self) -> Option<&Target> {
+        self.targets.iter().find(|target| match target.kind.as_slice() {
             &[ref kind] if kind == "lib" || kind == "proc-macro" => true,
             _ => false,
-        }
+        })
+    }
+
+    pub(crate) fn library_name(&self) -> Option<String> {
+        self.library_target().map(|target| target.name.replace("-", "_"))
```

Comment:
> I would actually expect it to be the other way around where library_target.name is the version with underscores.

That's not really possible. All the data in `src/utils/cargo_metadata.rs` is the deserialized output of the `cargo metadata --format-version=1` command, so that's what Cargo gives us. The method is to convert that data into the format we want.